File: app/src/main/java/tech/thdev/app/view/weekboxoffice/presenter/WeeklyBoxOfficeContract.java
package tech.thdev.app.view.weekboxoffice.presenter;

import android.content.Context;

import tech.thdev.app.adapter.movie.adapter.model.BoxOfficeAdapterContract;
import tech.thdev.app.data.source.MovieSearchRepository;
import tech.thdev.base.presenter.BasePresenter;
import tech.thdev.base.presenter.BaseView;

/**
 * Created by tae-hwan on 12/13/16.
 */

public interface WeeklyBoxOfficeContract {

    interface View extends BaseView {
        boolean isFinishing();

        void loadFail(String message);
    }

    interface Presenter extends BasePresenter<View> {

        void setSampleOneModel(BoxOfficeAdapterContract.Model model);

        void setSampleOneView(BoxOfficeAdapterContract.View view);

        void setMovieSearchRepository(MovieSearchRepository movieSearchRepository);

        void getWeeklyBoxOffice(Context context, String targetDate);
    }
}

File: app/src/main/java/tech/thdev/app/ui/weekboxoffice/presenter/WeeklyBoxOfficeContract.kt
package tech.thdev.app.ui.weekboxoffice.presenter

import android.content.Context
import tech.thdev.app.base.presenter.BasePresenter
import tech.thdev.app.base.presenter.BaseView

/**
 * Created by tae-hwan on 12/13/16.
 */
interface WeeklyBoxOfficeContract {
    interface View : BaseView {
        fun loadFail(messageRes: Int)
    }

    interface Presenter : BasePresenter<View> {
        fun getWeeklyBoxOffice(
            targetDate: String
        )
    }
}